### PR TITLE
Support Reset connection for VNG and Vwan

### PR DESCRIPTION
**Pull request by akshatkale**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/Network/Network/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/853
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/855
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by VeryEarly — CHANGES_REQUESTED** [review]
local SDK needed to be check into tools/localfeed.

There is another PR: https://github.com/Azure/azure-powershell/pull/14182

is going to check in local SDK, please make sure yours will be compatible with his.

**Comment by VeryEarly** [comment]
please sync with network-november branch, you might not need to check in the local SDK..

**Comment by akshatkale** [comment]
> please sync with network-november branch, you might not need to check in the local SDK..

@VeryEarly - Local SDK from https://github.com/Azure/azure-powershell/pull/14182 wasn't compatible with my changes so checked-in new Network SDK version.

File: src/Network/Network/Cortex/VpnConnection/ResetAzVpnSiteLinkConnectionCommand.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    using Microsoft.Azure.Commands.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.Management.Network;
+    using Microsoft.Azure.Management.Network.Models;
+    
+    [Cmdlet(
+        VerbsCommon.Reset,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VpnSiteLinkConnection",
+        DefaultParameterSetName = "ByName"),
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/VpnConnection/ResetAzVpnSiteLinkConnectionCommand.cs:14]:
```suggestion
        DefaultParameterSetName = "ByName",
        SupportsShouldProcess = true),
```

File: src/Network/Network/VirtualNetworkGatewayConnection/ResetAzureVirtualNetworkGatewayConnectionCommand.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.Management.Network;
+    using Microsoft.Azure.Management.Network.Models;
+    using Newtonsoft.Json;
+
+    [Cmdlet("Reset", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualNetworkGatewayConnection", DefaultParameterSetName = "ByName"), OutputType(typeof(void))]
```
Comment (VeryEarly) [inline src/Network/Network/VirtualNetworkGatewayConnection/ResetAzureVirtualNetworkGatewayConnectionCommand.cs:23]:
```suggestion
    [Cmdlet("Reset", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualNetworkGatewayConnection", DefaultParameterSetName = "ByName", SupportsShouldProcess = true), OutputType(typeof(void))]
```

File: src/Network/Network/Cortex/VpnConnection/ResetAzVpnSiteLinkConnectionCommand.cs (hunk 1 above)
Comment (akshatkale) [reply src/Network/Network/Cortex/VpnConnection/ResetAzVpnSiteLinkConnectionCommand.cs]: [Updated](https://github.com/Azure/azure-powershell/pull/14225/commits/cd04b1c4cc06f8a6a321139efa1037071f0ae76f#diff-29b5a4816afbb7a2db6e193d1ea4df4317c35a92a630490870336398d5927adbR15)

File: src/Network/Network/VirtualNetworkGatewayConnection/ResetAzureVirtualNetworkGatewayConnectionCommand.cs (hunk 2 above)
Comment (akshatkale) [reply src/Network/Network/VirtualNetworkGatewayConnection/ResetAzureVirtualNetworkGatewayConnectionCommand.cs]: [Updated](https://github.com/Azure/azure-powershell/pull/14225/commits/cd04b1c4cc06f8a6a321139efa1037071f0ae76f#diff-0d4fcb04a4e383d6b25257ec9d795c090e84df70b7b2bf97796549f3e23a13baR23)

**Comment by akshatkale** [comment]
Hi @VeryEarly, can we get this PR checked in today if everything looks good?

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**